src/shogun/multiclass/tree/C45ClassifierTree.cpp

```diff
@@ -141,10 +142,46 @@ bool C45ClassifierTree::train_machine(std::shared_ptr<Features> data)
 	feature_ids.range_fill();
 
 	set_root(C45train(data, m_weights, multiclass_labels(m_labels), feature_ids, 0));
-
+	// compute feature importances and normalize it
+	if (m_root)
+	{
+		m_feature_importances = SGVector<float64_t>(num_features);
+		m_feature_importances.zero();
+		compute_feature_importance(m_root);
+		float64_t total_num_sample = m_root->data.total_weight;
+		linalg::scale(
+		    m_feature_importances, m_feature_importances,
+		    1.0 / total_num_sample);
+		float64_t normalizer = linalg::sum(m_feature_importances);
+		if (normalizer > 0)
+			linalg::scale(
+			    m_feature_importances, m_feature_importances, 1.0 / normalizer);
+	}
 	return true;
 }
 
+void C45ClassifierTree::compute_feature_importance(
+    const std::shared_ptr<node_t>& node)
+{
+	auto childs = node->get_children();
+	m_feature_importances[node->data.attribute_id] +=
+	    node->data.impurity * node->data.total_weight;
+	for (const auto& child : childs)
```

Comment:
```suggestion
	#pragma omp parallel for shared(m_feature_importances)
	for (const auto& child : childs)
```

not so sure if its worth it though...